scripts/getChangelog.py

```diff
@@ -1,90 +1,587 @@
+            
+            If you're using bash or zsh, you can add the following line to your shell profile file (.bashrc, .bash_profile or .zshrc):
+            export GITHUB_API_TOKEN='{token}'
+            
+            If you're using Fish shell, you can add the following line to your config.fish file:
+            set -x GITHUB_API_TOKEN '{token}'
+            
+            After adding this line, you'll need to restart your terminal or run 'source ~/.bashrc' (or the appropriate file for your shell) for the changes to take effect.
+            
+            On Windows, you can set environment variables through the System Properties. Here's how:
+            1. Right-click on Computer on the desktop or in the Start menu.
+            2. Choose Properties.
+            3. Click on Advanced system settings.
+            4. Click on Environment Variables.
+            5. Click on New under the User or System variables sections.
+            6. Enter 'GITHUB_API_TOKEN' as the variable name and '{token}' as the variable value.
+            7. Click OK in all windows.
+            
+            Please note that setting environment variables this way will make them available in all future terminal sessions and scripts. Use with caution.
+            """))
+          else:
+            print("Ok, moving on.")
+          break
+      except urllib.error.HTTPError as error_message:
+        if error_message.code == 401:
+          handle_exception(Exception("Invalid GitHub API token, please try again."), data)
+          token = ""
+        else:
+          handle_exception(Exception("Issue executing GraphQL query"), data)
+          token = ""
+      except Exception as error_message:
+        handle_exception(error_message, data)
+        token = ""
+    return token
+
+  if not token:
+    token = get_token()
+  # Assuming authentication is successful, we will no longer check for/catch authentication errors.
+
+  # Define base_ref
+  if not base_ref:
+    print()
+    print("Now, what should be our point of reference?")
+    print("Please enter a release code or branch")
+    print("[default: latest (in other words: previous) GitHub release]")
+    base_ref = input('Base: ')
+
+  if base_ref == "latest" or not base_ref:
+    query = '''
+    query ($owner: String!, $repo: String!) {
+      repository(owner: $owner, name: $repo) {
+        latestRelease {
+          tagName
+        }
+      }
+    }
+    '''
+    while True:
+      try:
+        variables = {"owner": owner, "repo": repo}
+        response = client.execute(query, variables)
+        data = json.loads(response)
+        base_ref = data['data']['repository']['latestRelease']['tagName']
+        print("\nOkido, let's get the latest release (" + base_ref + ")!")
+        break
+      except Exception as e:
+        handle_exception(e, data)
+        print("Does your token have enough permissions?")
+        sys.exit(1)
```

Comment:
If you always either break or exit, this does not need to be in a `while True` loop